cgnsutilities/cgnsutilities.py

```diff
@@ -51,6 +52,48 @@ def __init__(self):
         self.name = "domain"
         self.cellDim = 3
 
+    def overwriteBCFamilyWithBC(self, familyName, newBCType, blocks=None, rstrip=True):
+        """
+        Overwrites all boundary conditions matching a given family name with a new boundary condition
+
+        This is useful because Pointwise specifies boundary conditions on CGNS grids in a way
+        that is incompatible with ADflow (family-defined) and these BC always need to be
+        overwritten.
+
+        Example
+        -------
+        from cgnsutilities.cgnsutilities import Grid, readGrid
+        grid = readGrid("pointwise_vol_grid.cgns")
+        grid.overwriteBCFamilyWithBC('oversetfamily', 'bcoverset', [0,1,2,4])
+        grid.writeToCGNS("pointwise_vol_grid_converted.cgns")
+
+        Inputs
+        ------
+        familyName : str
+            The BC family to overwrite
+        newBCType : str
+            The new boundary condition to apply
+        blocks : list of ints or None
+            The blocks to overwrite. Default None overwrites BC on all blocks.
+        rstrip : bool
+            It seems that the cgns reader always appends a large number of blank spaces
+            to the end of the family name. These need to be stripped. Default True.
+        """
+        if newBCType not in BC.keys():
+            raise ValueError("New BC type ", newBCType, " is not in the cgnsutilities list of boundary conditions.")
+        nBCOverwritten = 0
+        for iBlk, block in enumerate(self.blocks):
+            if blocks is None or iBlk in blocks:
```

Comment:
If the user specifies a block that doesn't exist then it should throw some kind of error. 
Especially because the block numbering here is the index of the block in the grid's list of blocks and not some property of the grid's definition. 
If your meshing software wasn't zero based like python then you might try to change the BC of a block that doesn't exist.

Reply:
maybe something like. 
```
if blocks is None:
    blocks = range(len(self.blocks))

for idx_blk  in blocks:
    if idx_blk not in range(len(self.blocks)): 
         raise error(f"block {idx_blk} specified, but does not exist in the grid")
```

Reply:
can't we just add a comment in the docstring that the indices should be zero-based? Simply warning for indices that don't exist might miss issues where the users are using a 1-based indexing but still get a match on the grid

Reply:
Good point, Lets do that instead